clippy_lints/src/methods/open_options.rs

```diff
@@ -82,97 +99,52 @@ fn get_open_options(cx: &LateContext<'_>, argument: &Expr<'_>, options: &mut Vec
+        if options.insert(option.clone(), arg.clone()).is_some() {
+            span_lint(
+                cx,
+                NONSENSICAL_OPEN_OPTIONS,
+                span,
+                &format!("the method `{}` is called more than once", &option),
+            );
+        }
+    }
 
-    for option in options {
-        match *option {
-            (OpenOption::Create, arg) => {
-                if create {
-                    span_lint(
-                        cx,
-                        NONSENSICAL_OPEN_OPTIONS,
-                        span,
-                        "the method `create` is called more than once",
-                    );
-                } else {
-                    create = true;
-                }
-                create_arg = create_arg || (arg == Argument::True);
-            },
-            (OpenOption::Append, arg) => {
-                if append {
-                    span_lint(
-                        cx,
-                        NONSENSICAL_OPEN_OPTIONS,
-                        span,
-                        "the method `append` is called more than once",
-                    );
-                } else {
-                    append = true;
-                }
-                append_arg = append_arg || (arg == Argument::True);
-            },
-            (OpenOption::Truncate, arg) => {
-                if truncate {
-                    span_lint(
-                        cx,
-                        NONSENSICAL_OPEN_OPTIONS,
-                        span,
-                        "the method `truncate` is called more than once",
-                    );
-                } else {
-                    truncate = true;
-                }
-                truncate_arg = truncate_arg || (arg == Argument::True);
-            },
-            (OpenOption::Read, arg) => {
-                if read {
-                    span_lint(
-                        cx,
-                        NONSENSICAL_OPEN_OPTIONS,
-                        span,
-                        "the method `read` is called more than once",
-                    );
-                } else {
-                    read = true;
-                }
-                read_arg = read_arg || (arg == Argument::True);
-            },
-            (OpenOption::Write, arg) => {
-                if write {
-                    span_lint(
-                        cx,
-                        NONSENSICAL_OPEN_OPTIONS,
-                        span,
-                        "the method `write` is called more than once",
-                    );
-                } else {
-                    write = true;
-                }
-                write_arg = write_arg || (arg == Argument::True);
-            },
+    if let (Some(Argument::Set(true)), Some(Argument::Set(true))) =
+        (options.get(&OpenOption::Read), options.get(&OpenOption::Truncate))
+    {
+        if options.get(&OpenOption::Write).unwrap_or(&Argument::Set(false)) == &Argument::Set(false) {
```

Comment:
Nit: the `OpenOption::Write` check can be merged with the other two checks, either with just `&&` or
```suggestion
    if let (Some(Argument::Set(true)), Some(Argument::Set(true)), None | Some(Argument::Set(false))) = (
        options.get(&OpenOption::Read),
        options.get(&OpenOption::Truncate),
        options.get(&OpenOption::Write),
    ) {
```

Reply:
Done, I have actually changed to follow the more-common style by using `if_chain!`

Reply:
> Done, I have actually changed to follow the more-common style by using `if_chain!`

New code should prefer using if let chains over the `if_chain!` macro, so e.g.
```rs
if_chain! {
  if let Some((Argument::Set(true), _)) = options.get(&OpenOption::Read);
  if let Some((Argument::Set(true), _)) = options.get(&OpenOption::Truncate);
  if let None | Some((Argument::Set(false), _)) = options.get(&OpenOption::Write);
  then {
    ...
  }
}
```
becomes
```rs
if let Some((Argument::Set(true), _)) = options.get(&OpenOption::Read)
    && let Some((Argument::Set(true), _)) = options.get(&OpenOption::Truncate)
    && let None | Some((Argument::Set(false), _)) = options.get(&OpenOption::Write)
{
  ..
}
```

Reply:
Did not know that syntax, fixed. Thanks

Reply:
It's still unstable (requires the `let_chains` feature), but we want to help test it :)